internal/providers/github/entities.go

```diff
@@ -45,13 +55,145 @@ func (c *GitHub) SupportsEntity(entType minderv1.Entity) bool {
+		return nil, fmt.Errorf("error joining webhook URL: %w", err)
+	}
+
+	parsedWebhookURL, err := url.Parse(webhookURL)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing webhook URL: %w", err)
+	}
+
+	repoNameP := props.GetProperty(ghprop.RepoPropertyName)
+	if repoNameP == nil {
+		return nil, errors.New("repo name property not found")
+	}
+
+	repoOwnerP := props.GetProperty(ghprop.RepoPropertyOwner)
+	if repoOwnerP == nil {
+		return nil, errors.New("repo owner property not found")
+	}
+
+	repoName := repoNameP.GetString()
+	repoOwner := repoOwnerP.GetString()
+
+	// If we have an existing hook for same repo, delete it
+	if err := c.cleanupStaleHooks(ctx, repoOwner, repoName, parsedWebhookURL.Host); err != nil {
+		return nil, fmt.Errorf("error cleaning up stale hooks: %w", err)
+	}
+
+	// Attempt to register new webhook
+	secret, err := c.webhookConfig.GetWebhookSecret()
+	if err != nil {
+		return nil, fmt.Errorf("error getting webhook secret: %w", err)
+	}
+	ping := c.webhookConfig.ExternalPingURL
+
+	jsonCT := "json"
+	newHook := &github.Hook{
+		Config: &github.HookConfig{
+			URL:         &webhookURL,
+			ContentType: &jsonCT,
+			Secret:      &secret,
+		},
+		PingURL: &ping,
+		Events:  targetedEvents,
+	}
+
+	webhook, err := c.CreateHook(ctx, repoOwner, repoName, newHook)
+	if err != nil {
+		return nil, fmt.Errorf("error creating hook: %w", err)
+	}
+
+	whprops, err := properties.NewProperties(map[string]any{
+		ghprop.RepoPropertyHookUiid: hookUUID,
+		ghprop.RepoPropertyHookId:   webhook.GetID(),
+		ghprop.RepoPropertyHookUrl:  webhook.GetURL(),
+		ghprop.RepoPropertyHookName: webhook.GetName(),
+		ghprop.RepoPropertyHookType: webhook.GetType(),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error creating webhook properties: %w", err)
+	}
+
+	return props.Merge(whprops), nil
 }
 
 // DeregisterEntity implements the Provider interface
-func (_ *GitHub) DeregisterEntity(_ context.Context, _ minderv1.Entity, _ *properties.Properties) error {
-	// TODO: implement
+func (c *GitHub) DeregisterEntity(ctx context.Context, entityType minderv1.Entity, props *properties.Properties) error {
+	// We only need explicit registration steps for repositories
+	// The rest of the entities are originated from them.
+	if entityType != minderv1.Entity_ENTITY_REPOSITORIES {
+		return nil
+	}
+
+	repoNameP := props.GetProperty(ghprop.RepoPropertyName)
+	if repoNameP == nil {
+		return errors.New("repo name property not found")
+	}
+
+	repoOwnerP := props.GetProperty(ghprop.RepoPropertyOwner)
+	if repoOwnerP == nil {
+		return errors.New("repo owner property not found")
+	}
+
+	hookIDP := props.GetProperty(ghprop.RepoPropertyHookId)
+	if hookIDP == nil {
+		return errors.New("hook ID property not found")
+	}
+
+	repoName := repoNameP.GetString()
+	repoOwner := repoOwnerP.GetString()
+	hookID := hookIDP.GetInt64()
+
+	err := c.DeleteHook(ctx, repoOwner, repoName, hookID)
+	if err != nil {
+		return fmt.Errorf("error deleting hook: %w", err)
```

Comment:
Should we treat this error that calls upstream and the errors where we don't have the required properties differently? e.g. the upstream call error could be transient and the user might just want to retry but if the properties are not there, then there's no point in retrying, shouldn't we just proceed with deleting the entity in that case?